File: 2025-09-22/digitsVsLetters.js
function digitsOrLetters(str) {
  let digits = 0;
  let letters = 0;

  for (let char of str) {
    if (/[0-9]/.test(char))
      digits ++;
    
    if (/[a-zA-Z]/.test(char))
      letters ++;
  }

  return digits > letters ? "digits" 
    : digits < letters ? "letters" 
    : "tie";
}

console.log(digitsOrLetters("abc123"));
console.log(digitsOrLetters("a1b2c3d"));
console.log(digitsOrLetters("1a2b3c4"));
console.log(digitsOrLetters("abc123!@#DEF"));
console.log(digitsOrLetters("H3110 W0R1D"));
console.log(digitsOrLetters("P455W0RD"));

File: 2025-09-22/digits_vs_letters.py
def digits_or_letters(s):
    import re
    digits = 0
    letters = 0

    for char in s:
        if re.match(r"[0-9]", char):
            digits += 1

        if re.match(r"[a-zA-Z]", char):
            letters += 1

    return "digits" if digits > letters \
        else "letters" if digits < letters \
        else "tie"

print(digits_or_letters("abc123"))
print(digits_or_letters("a1b2c3d"))
print(digits_or_letters("1a2b3c4"))
print(digits_or_letters("abc123!@#DEF"))
print(digits_or_letters("H3110 W0R1D"))
print(digits_or_letters("P455W0RD"))